src/gluonts/dataset/split/splitter.py

```diff
@@ -71,47 +71,70 @@ class Config:
     feat_dynamic_real: List[pd.Series] = []
 
     @classmethod
-    def from_time_series_item(
-        cls, item: TimeSeriesItem, freq: Optional[str] = None
+    def from_data_entry(
+        cls, item: DataEntry, freq: Optional[str] = None
     ) -> "TimeSeriesSlice":
         if freq is None:
-            freq = item.start.freq
+            freq = item["start"].freq
 
         index = pd.date_range(
-            start=item.start, freq=freq, periods=len(item.target)
+            start=item["start"], freq=freq, periods=len(item["target"])
         )
 
+        item_dynamic_cat = (
+            item["feat_dynamic_cat"] if "feat_dynamic_cat" in item else []
+        )
         feat_dynamic_cat = [
-            pd.Series(cat, index=index) for cat in item.feat_dynamic_cat
+            pd.Series(cat, index=index) for cat in item_dynamic_cat
         ]
 
+        item_dynamic_real = (
+            item["feat_dynamic_real"] if "feat_dynamic_real" in item else []
+        )
         feat_dynamic_real = [
-            pd.Series(real, index=index) for real in item.feat_dynamic_real
+            pd.Series(real, index=index) for real in item_dynamic_real
         ]
 
+        feat_static_cat = (
+            list(item["feat_static_cat"]) if "feat_static_cat" in item else []
+        )
+
+        feat_static_real = (
+            list(item["feat_static_real"])
+            if "feat_static_real" in item
+            else []
+        )
+
         return TimeSeriesSlice(
-            target=pd.Series(item.target, index=index),
-            item=item.item,
-            feat_static_cat=item.feat_static_cat,
-            feat_static_real=item.feat_static_real,
+            target=pd.Series(item["target"], index=index),
+            item=item["item"],
+            feat_static_cat=feat_static_cat,
+            feat_static_real=feat_static_real,
             feat_dynamic_cat=feat_dynamic_cat,
             feat_dynamic_real=feat_dynamic_real,
         )
 
-    def to_time_series_item(self) -> TimeSeriesItem:
-        return TimeSeriesItem(
-            start=self.start,
-            target=self.target.values,
-            item=self.item,
-            feat_static_cat=self.feat_static_cat,
-            feat_static_real=self.feat_static_real,
-            feat_dynamic_cat=[
+    def to_data_entry(self) -> DataEntry:
+        ret = {
+            "start": self.start,
+            "item": self.item,
+            "target": self.target.values,
+        }
+
+        if len(self.feat_static_cat) > 0:
```

Comment:
In Python, you almost never need to use `len` to check whether an element has entries:

```suggestion
        if self.feat_static_cat:
```